### PR TITLE
Replace perl by rest api for radius accounting

### PR DESCRIPTION
**Pull request by fdurand**
# Description

Use rest call for radius accounting instead of perl
# Impacts

RADIUS accounting workflow
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Use rest call for radius accounting instead of perl

### CODE REVIEW

File: lib/pf/api.pm [hunk 1]
```diff
@@ -1327,22 +1327,45 @@ sub radius_rest_switch_authorize :Public :RestPath(/radius/rest/switch/authorize
...
 sub handle_accounting_metadata : Public {
     my ($class, %RAD_REQUEST) = @_;
     my $logger = pf::log::get_logger();
     $logger->debug("Entering handling of accounting metadata");
     my $client = pf::client::getClient();
 
+    my $return = [ $RADIUS::RLM_MODULE_OK, ('Reply-Message' => "Accounting OK") ];
     my $mac = pf::util::clean_mac($RAD_REQUEST{'Calling-Station-Id'});
-    if ($RAD_REQUEST{'Acct-Status-Type'} eq 'Start') {
+    if ($RAD_REQUEST{'Acct-Status-Type'} == 1) {
```
Comment (julsemaan) [inline lib/pf/api.pm]: constant for this

File: lib/pf/api.pm [hunk 2]
```diff
@@ -1327,22 +1327,45 @@ sub radius_rest_switch_authorize :Public :RestPath(/radius/rest/switch/authorize
...
         #
         # Updating location log in on initial ('Start') accounting run.
         #
         $logger->info("Updating locationlog from accounting request");
         $client->notify("radius_update_locationlog", %RAD_REQUEST);
     }
 
-    if ($RAD_REQUEST{'Acct-Status-Type'} ne 'Stop'){
+    if ($RAD_REQUEST{'Acct-Status-Type'} == 2){
```
Comment (julsemaan) [inline lib/pf/api.pm]: constant for that

File: lib/pf/api.pm [hunk 3]
```diff
@@ -1352,7 +1375,16 @@ sub handle_accounting_metadata : Public {
             pf::log::get_logger->debug("Not handling iplog update because we're not configured to do so on accounting packets.");
         }
     }
-
+    if ($RAD_REQUEST{'Acct-Status-Type'} == 2 || $RAD_REQUEST{'Acct-Status-Type'} == 3 ) {
```
Comment (julsemaan) [inline lib/pf/api.pm]: constants everywhere

**Review by julsemaan — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
Bump

**Pull request merged**